### PR TITLE
Add group functions

### PR DESCRIPTION
**Pull request by vchuravy**
x-ref: #277

### CODE REVIEW

File: src/group.jl [hunk 1]
```diff
@@ -0,0 +1,111 @@
+@mpi_handle Group
+
+const GROUP_NULL = _Group(MPI_GROUP_NULL)
+const GROUP_EMPTY = _Group(MPI_GROUP_EMPTY)
+
+Group() = Group(GROUP_NULL.val)
+
+function free(group::Group)
+    if group.val != GROUP_NULL.val
+        @mpichk ccall((:MPI_Comm_free, libmpi), Cint, (Ptr{MPI_Group},), group)
```
Comment (simonbyrne) [inline src/group.jl]: I've started pasting the C signatures (from the MPI spec) above the ccalls: I've found it useful to check we're passing arguments correctly.

File: src/group.jl [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    @mpichk ccall((:MPI_Group_compare, libmpi), Cint,
+        (MPI_Group, MPI_Group, Ptr{Cint}), group1, group2, result)
+    Int(result[])
+end
+
+function Group_difference(group1::Group, group2::Group)
+    newgroup = Group()
+    @mpichk ccall((:MPI_Group_difference, libmpi), Cint,
+        (MPI_Group, MPI_Group, Ptr{MPI_Group}), group1, group2, newgroup)
+    return newgroup
```
Comment (simonbyrne) [inline src/group.jl]: should this one and below have finalizers?

Comment (vchuravy) [reply src/group.jl]: Ah yes they should

**Comment by lucifer1004** [comment]
Will this PR ever be merged?

**Comment by vchuravy** [comment]
I have not found the need for this in my own work, so I would be happy for someone who is interested in this to take this over.

**Comment by carstenbauer** [comment]
I would also like to see this merged! I hope that I'll find some time in the next few weeks to take this one over. (Of course, I'd be happy if someone beats me to it 😄)

**Comment by carlocastoldi** [comment]
I needed this and so I rebased the pull request to `master`, fixed little things and added support to `MPICH` ([HPE-MPI](https://buy.hpe.com/us/en/software/high-performance-computing-software/high-performance-computing-software/hpe-message-passing-interface-mpi/p/1010144155), that's what my workplace uses).

Should I try opening a new pull request?

**Comment by simonbyrne** [comment]
> Should I try opening a new pull request?

Yes please!

**Comment by simonbyrne** [comment]
Now done in #521

**Pull request closed without merging**